File: GrpProj1/BlackjackJS/Card.js
/*
	File: Card.js 
	Author: Kevin Romero
	Class: CSC17B Team Blackjack
	Purpose: Card object for Blackjack
	Date: 10/16/17
*/


//Enums
var SUIT = Object.freeze({"SPADES":1, "HEARTS":2, "CLUBS":3,
										"DIAMONDS":4});
var FACE = Object.freeze({"ACE":1, "TWO":2, "THREE":3, "FOUR":4, "FIVE":5,
						"SIX":6, "SEVEN":7, "EIGHT":8, "NINE":9,
						"TEN":10,"JACK":11,"QUEEN":12,"KING":13,"fCount":14});


//Card Class
function Card(face,suit){
	//member variables
	this.m_suit;
	this.m_face;
	
	//constructor
	this.setCard(face,suit);
};

//Constructors
Card.prototype.setCard=function(face,suit){
	this.setFace(face);
	this.setSuit(suit);
};

/*Card.prototype.setCardI=function(i){
	this.setSuit(i/FACE.fCount);
	this.setFace(i%FACE.fCount);
}*/

//Mutators
Card.prototype.setFace=function(face){
	this.m_face=face;
};

Card.prototype.setSuit=function(suit){
	this.m_suit=suit;
};

//Accessors
Card.prototype.getSetCard=function(){
	return this.toStringFace(this.getFace())+" of "+this.toStringSuit(this.getSuit());
};

Card.prototype.getCardVal=function(face){
	switch(face){
		case FACE.JACK:
			return 10;
		case FACE.QUEEN:
			return 10;
		case FACE.KING:
			return 10;
		default: return face;
	}
};

Card.prototype.getGivenCard=function(face,suit){
	return this.toStringFace(face)+" of "+this.toStringSuit(suit);
};


Card.prototype.getFace=function(){
	return this.m_face;
};

Card.prototype.getSuit=function(){
	return this.m_suit;
};

//toString functions
Card.prototype.toStringSuit=function(suit){
	switch(suit){
		case 1:
			return "SPADES";
		case 2:
			return "HEARTS";
		case 3:
			return "CLUBS";
		case 4:
			return "DIAMONDS";
		default: return "ERROR";
	}
};

Card.prototype.toStringFace=function(face){
	switch(face){
		case 1:
			return "ACE";
		case 2:
			return "TWO";
		case 3:
			return "THREE";
		case 4:
			return "FOUR";
		case 5:
			return "FIVE";
		case 6:
			return "SIX";
		case 7:
			return "SEVEN";
		case 8:
			return "EIGHT";
		case 9:
			return "NINE";
		case 10:
			return "TEN";
		case 11:
			return "JACK";
		case 12:
			return "QUEEN";
		case 13:
			return "KING";
		default:
			return "ERROR";
	}
};

File: GrpProj1/Blackjack code in C++/Blackjack-master/Card.cpp
/*
 * File: Card.cpp
 * Purpose: Implementation of Card Class
 * Date: June 28, 2017
 */

//System Libraries
#include <iostream>

//User Libraries
#include "Card.h"

//Default Constructor
Card::Card(){
    
}

//Constructor
Card::Card(FACE const& f, SUIT const &s){
    this->setFace(f);
    this->setSuit(s);
}

//Constructor
Card::Card(int i){
    this->setSuit(static_cast<SUIT>(i/fCount));
    this->setFace(static_cast<FACE>(i%fCount));
}

//Get card number value
int Card::getCardVal(FACE const& f){
    switch(static_cast<int>(f)){
        case(static_cast<int>(FACE::JACK)):  return static_cast<int>(FACE::TEN);
        case(static_cast<int>(FACE::QUEEN)): return static_cast<int>(FACE::TEN);
        case(static_cast<int>(FACE::KING)):  return static_cast<int>(FACE::TEN);
        default:                             return static_cast<int>(f);
    }
}

//return given card as string
std::string Card::getGivenCard(const FACE& f, const SUIT& s){
    std::string str;
    str = this->toString(f) + " of " + this->toString(s);
    return str;
}

//Implement toString
std::string Card::toString(FACE const&f){
    switch(f){
        case FACE::ACE:     return "Ace";
        case FACE::TWO:     return "Two";
        case FACE::THREE:   return "Three";
        case FACE::FOUR:    return "Four";
        case FACE::FIVE:    return "Five";
        case FACE::SIX:     return "Six";
        case FACE::SEVEN:   return "Seven";
        case FACE::EIGHT:   return "Eight";
        case FACE::NINE:    return "Nine";
        case FACE::TEN:     return "Ten";
        case FACE::JACK:    return "Jack";
        case FACE::QUEEN:   return "Queen";
        case FACE::KING:    return "King";
        default: throw std::logic_error("invalid face");
   }
}

std::string Card::toString(SUIT const&s){
    switch(s){
        case SUIT::CLUBS:       return "Clubs";
        case SUIT::DIAMONDS:    return "Diamonds";
        case SUIT::HEARTS:      return "Hearts";
        case SUIT::SPADES:      return "Spades";
        default: throw std::logic_error("invalid suit");
    }

}